platform/android/MapLibreAndroid/src/vulkan/java/org/maplibre/android/maps/renderer/MapRenderer.java

```diff
@@ -0,0 +1,202 @@
+import org.maplibre.android.maps.renderer.textureview.TextureViewMapRenderer;
+
+/**
+ * The {@link MapRenderer} encapsulates the render thread.
+ * <p>
+ * Performs actions on the render thread to manage the resources and
+ * render on the one end and acts as a scheduler to request work to
+ * be performed on the render thread on the other.
+ */
+@Keep
+public abstract class MapRenderer implements MapRendererScheduler {
+
+  static {
+    LibraryLoader.load();
+  }
+
+  private static final String TAG = "Mbgl-MapRenderer";
+
+  // Holds the pointer to the native peer after initialisation
+  private long nativePtr = 0;
+  private double expectedRenderTime = 0;
+  private MapLibreMap.OnFpsChangedListener onFpsChangedListener;
+
+  public static MapRenderer create(MapLibreMapOptions options, @NonNull Context context, Runnable initCallback) {
+
+    MapRenderer renderer = null;
+    String localFontFamily = options.getLocalIdeographFontFamily();
+
+    if (options.getTextureMode()) {
+      TextureView textureView = new TextureView(context);
+      boolean translucentSurface = options.getTranslucentTextureSurface();
+      renderer = new TextureViewMapRenderer(context, textureView, localFontFamily, translucentSurface) {
+        @Override
+        protected void onSurfaceCreated(Surface surface) {
+          initCallback.run();
+          super.onSurfaceCreated(surface);
+        }
+      };
+    } else {
+      MapLibreSurfaceView surfaceView = new MapLibreSurfaceView(context);
+      surfaceView.setZOrderMediaOverlay(options.getRenderSurfaceOnTop());
+      renderer = new SurfaceViewMapRenderer(context, surfaceView, localFontFamily) {
+        @Override
+        public void onSurfaceCreated() {
+          initCallback.run();
+          super.onSurfaceCreated();
+        }
+      };
+    }
+
+    return renderer;
+  }
+
+  public MapRenderer(@NonNull Context context, String localIdeographFontFamily) {
+    float pixelRatio = context.getResources().getDisplayMetrics().density;
+
+    // Initialise native peer
+    nativeInitialize(this, pixelRatio, localIdeographFontFamily);
+  }
+
+  public abstract View getView();
+
+  public void onStart() {
+    // Implement if needed
+  }
+
+  public void onPause() {
+    // Implement if needed
+  }
+
+  public void onResume() {
+    // Implement if needed
+  }
+
+  public void onStop() {
+    // Implement if needed
+  }
+
+  public void onDestroy() {
+    // Implement if needed
+  }
+
+  public void setOnFpsChangedListener(MapLibreMap.OnFpsChangedListener listener) {
+    onFpsChangedListener = listener;
+  }
+
+  @CallSuper
+  protected void onSurfaceCreated(Surface surface) {
+    nativeOnSurfaceCreated(surface);
+  }
+
+  @CallSuper
+  protected void onSurfaceChanged(int width, int height) {
+    nativeOnSurfaceChanged(width, height);
+  }
+
+  @CallSuper
+  protected void onSurfaceDestroyed() {
+    nativeOnSurfaceDestroyed();
+  }
+
+  @CallSuper
+  protected void onDrawFrame() {
```

Comment:
Duplicated. I feel like this should be in a common base class.

Reply:
Moved `MapRenderer` to the main source set and generated factory classes for implementations. This will also help when/if we decide to merge the source sets and have a build with both backends available